### DISCUSSION TITLE
KEYCLOAK SPI: Keycloak Provider is null when trying to get data from db

### DISCUSSION THREAD

**Discussion by Naouf3l** [keycloak/keycloak#8474] [Q&A]
I have an issue on creating an spi to extend keycloak and create a new entity.
I followed the zonaut example(https://github.com/zonaut/keycloak-extensions/tree/master/provider-domain), and it works to add a new entity.

But when I try to get or save entities from the keycloak database using a keycloak provider as the keycloak example I have an issue. The provider is null.
For this case I followed the keycloak example:
https://github.com/keycloak/keycloak/tree/master/examples/providers/domain-extension

Here is my repo if you want to look at the source code:
https://github.com/Naouf3l/kc-extension

Thanks.

**Comment by stianst** [comment]
I would suggest starting with one of our quickstarts. Take a look at https://github.com/keycloak/keycloak-quickstarts/tree/latest/user-storage-jpa for example.
